lexer/lexer.go

```diff
@@ -68,6 +68,10 @@ func (l *Lexer) NextToken() *token.Token {
 			// so we normalize := like it didn't exist.
 			return token.ConstantTokenChar2(ch, nextChar)
 		}
+		if nextChar == '>' && ch == '=' {
```

Comment:
Maybe you could invert


```suggestion
		if ch == '=' && nextChar == '>' {
```

Or add a comment


```suggestion
		// detecting lamda function ( '=>' )
		if nextChar == '>' && ch == '=' {
```

Or something like that

Reply:
I added a comment, the test is in the order it is because ch is already likely to be = (or ! ...) so I want the most specific first and no retest unless needed